string model_name
string[] param_names
float64[] model_params
DataPoint[] seg_stats
int32 first_point
int32 last_point
